### PR TITLE
Implement Unix file regularity check (#20448)

### PR DESCRIPTION
**Pull request by a-mr**
Implements #20448.

### CODE REVIEW

**Comment by ringabout** [comment]
What about `walkPattern` and `walkPattern`?

**Review by ringabout — COMMENTED** [review]
What about `walkPattern` and `walkFiles`? There is also a new module called `std/dirs` which is a type safe version of `std/private/osdirs` which should be patched too.

**Comment by a-mr** [comment]
A user can just combine `walkPattern/walkFiles` with a call to `getFileInfo` and checking field `isRegular`. 

This was a conscious design decision :-). `walkPattern/walkFiles` use Unix `glob` function for implementation, which does not return info on file type. The point of this PR is in that current syscalls in `walkDir/getFileInfo` already return the needed information, so this regularity check can be done without any overhead. (Also note that `walkPattern/walkFiles` don't have options `relative` and `checkDir` implemented.) TBH **my opinion** is that `walkPattern/walkFiles` are redundant and had better be deprecated.

- [x] std/dirs — done

File: lib/pure/os.nim [hunk 1]
```diff
@@ -1095,6 +1095,8 @@ type
     creationTime*: times.Time         ## Time file was created. Not supported on all systems!
     blockSize*: int                   ## Preferred I/O block size for this object.
                                       ## In some filesystems, this may vary from file to file.
+    isRegular*: bool                  ## Is file regular? (on Unix some "files"
+                                      ## can be non-regular like FIFOs, devices)
 
 template rawToFormalFileInfo(rawInfo, path, formalInfo): untyped =
```
Comment (Varriount) [inline lib/pure/os.nim:1101]: - On \*nix, are symlinks, etc. considered "regular files"?
- Windows needs to be handled. If the above question is `true`, then `isRegular` can just always be set to `true`.
- If it's not too much trouble, I think it's time for `rawToFormalFileInfo` to be split into 2 different templates (one for each platform).

**Comment by ringabout** [comment]
> TBH my opinion is that walkPattern/walkFiles are redundant and had better be deprecated.

It can be removed from `std/dirs` which is a new module.

File: lib/pure/os.nim (hunk 1 above)
Comment (a-mr) [reply lib/pure/os.nim:1101]: > On *nix, are symlinks, etc. considered "regular files"?

Not really, they are a separate type of file (even links to directories).  But here `isRegular` can be both `true` or `false` depending on target. I'll think about documenting that more thorougly.

> Windows needs to be handled. If the above question is true, then isRegular can just always be set to true.

Thank you. May be an omission, I'll check that.

**Pull request merged**